integration/src/main/java/scripts/utils/GcsBucketAccessTester.java

```diff
@@ -154,17 +272,34 @@ public interface TestFunction {
   }
 
   private boolean doWithOptionalWait(TestFunction function) throws Exception {
+    if (needToWait) {
+      return doWait(function);
+    }
+    return doNoWait(function);
+  }
+
+  private boolean doWait(TestFunction function) throws Exception {
     try {
-      if (needToWait) {
-        ClientTestUtils.getWithRetryOnException(function::apply);
-        needToWait = false;
-      } else {
-        function.apply();
+      ClientTestUtils.getWithRetryOnException(function::apply);
+      return true;
+    } catch (StorageException e) {
+      if (e.getCode() == HttpStatusCodes.STATUS_CODE_FORBIDDEN) {
+        assertEquals(HttpStatusCodes.STATUS_CODE_FORBIDDEN, e.getCode());
```

Comment:
this assertion will always be true given the branch above

Reply:
Removed the assert.